### PR TITLE
Add gcloud build configuration

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
/kind test
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

File: hack/release.sh [hunk 1]
```diff
@@ -41,12 +41,14 @@ Environments:
...
     REGISTRY                    container registry without repo name (default: quay.io/external_storage)
     VERSION                     if set, use given version as image tag
+    PULL_BASE_REF               if set, detect version from this git ref instead of "git describe"
     DOCKER_CONFIG               optional docker config location
     CONFIRM                     set this to skip confirmation
     ALLOW_UNSTABLE              by default, only master branch and tags that matches v<major>.<minor>.<patch> format are allowed, set this to skip this check (debug only)
     ALLOW_DIRTY                 by default, git repo must be clean, set this to skip this check (debug only)
     ALLOW_OVERRIDE              by default, stable image is not allowed to override, set this to skip (debug only)
     SKIP_BUILD                  set this to skip build phase (debug only)
+    SKIP_PUSH_LATEST            set this to skip pushing the latest stable image as the latest image
```
Comment (cofyc) [inline hack/release.sh:51]: [current prow job](https://github.com/kubernetes/test-infra/blob/2b47381ded875b75626ae9d5e3c3d6e5839a51e9/config/jobs/image-pushing/k8s-staging-sig-storage.sh) does not include `.git` directory, I add this to skip checking if the tag is the latest.

the `latest` tag is not recommended in production, I think it's fine not pushing it.

File: cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+steps:
+- name: gcr.io/k8s-testimages/gcb-docker-gcloud:v20200421-a2bf5f8
+  entrypoint: hack/release.sh
+  env:
+  - REGISTRY=gcr.io/${_STAGING_PROJECT}
+  - ALL_ARCH=${_ALL_ARCH}
+  - ALLOW_DIRTY=yes
+  - ALLOW_UNSTABLE=yes
+  - CONFIRM=yes
+  - SKIP_PUSH_LATEST=yes
```
Comment (msau42) [inline cloudbuild.yaml:19]: How will we support the canary job? https://k8s-testgrid.appspot.com/sig-storage-local-static-provisioner#master-gce-lastest

Comment (msau42) [reply cloudbuild.yaml:19]: Oh I see canary is different from latest

**Pull request merged**